src/processing/vrf.rs

```diff
@@ -0,0 +1,138 @@
+extern crate diesel;
+extern crate dotenv;
+
+use bigdecimal::BigDecimal;
+use chrono::Utc;
+use deadpool::managed::Pool;
+use diesel::dsl::max;
+use diesel::ExpressionMethods;
+use diesel::OptionalExtension;
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+use diesel_async::AsyncPgConnection;
+use diesel_async::RunQueryDsl;
+use starknet::core::types::Felt;
+
+use crate::config::get_config;
+use crate::constants::{
+    VRF_BALANCE, VRF_REQUESTS_COUNT, VRF_TIME_SINCE_LAST_HANDLE_REQUEST,
+    VRF_TIME_SINCE_OLDEST_REQUEST_IN_PENDING_STATUS,
+};
+use crate::diesel::QueryDsl;
+use crate::error::MonitoringError;
+use crate::models::VrfRequest;
+use crate::monitoring::get_on_chain_balance;
+use crate::schema::vrf_requests::dsl as vrf_dsl;
+
+#[derive(Debug, Copy, Clone)]
+#[allow(dead_code)]
+enum VrfStatus {
+    Uninitialized,
+    Received,
+    Fulfilled,
+    Cancelled,
+    OutOfGas,
+    Refunded,
+}
+
+impl From<VrfStatus> for BigDecimal {
+    fn from(val: VrfStatus) -> Self {
+        BigDecimal::from(val as i32)
+    }
+}
+
+pub async fn check_vrf_balance(vrf_address: Felt) -> Result<(), MonitoringError> {
+    let config = get_config(None).await;
+    let balance = get_on_chain_balance(vrf_address).await?;
+    let network_env = &config.network_str();
+    VRF_BALANCE.with_label_values(&[network_env]).set(balance);
+    Ok(())
+}
+
+pub async fn check_vrf_request_count(
+    pool: Pool<AsyncDieselConnectionManager<AsyncPgConnection>>,
+) -> Result<(), MonitoringError> {
+    let mut conn = pool.get().await.map_err(MonitoringError::Connection)?;
+
+    let networks: Vec<String> = vrf_dsl::vrf_requests
```

Comment:
remove these, we only have one exporter per network. you can get the exporter's network using
```rust
let config = get_config(None).await;
config.network()
```

Reply:
done